### PR TITLE
Provide relevant block data in `ExtendVote`

### PR DESCRIPTION
**Pull request by sergio-mena**
Contributes to #12

Some use cases require `RequestExtendBlock` to contain the block (rather than just its hash) that the precommit message containing the vote extension will refer to. This PR extends the fields of `RequestExtendBlock` and populates them when calling `ExtendVote` ABCI call.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by lasarojc — COMMENTED** [review]
LGTM
Good work making the interface consistent with ProcessProposal

File: state/execution.go [hunk 1]
```diff
@@ -313,10 +313,27 @@ func (blockExec *BlockExecutor) ApplyBlock(
 	return state, nil
 }
 
-func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) ([]byte, error) {
+func (blockExec *BlockExecutor) ExtendVote(
+	ctx context.Context,
+	vote *types.Vote,
+	block *types.Block,
+	state State,
+) ([]byte, error) {
```
Comment (thanethomson) [inline state/execution.go:321]: We'll need a changelog entry to highlight this Go API-breaking change.

File: .changelog/unreleased/breaking-changes/1270-executor_extend_vote.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+`[state]` Signature of `ExtendVote` changed in `BlockExecutor`.
```
Comment (thanethomson) [inline .changelog/unreleased/breaking-changes/1270-executor_extend_vote.md:1]:
```suggestion
- `[state]` Signature of `ExtendVote` changed in `BlockExecutor`.
```

**Review by thanethomson — APPROVED** [review]
One minor nit, but otherwise good!

**Pull request merged**